From 6eae18ba760a2bee043e22215a2202577ea0fc43 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 14 Mar 2019 03:55:50 -0700
Subject: [PATCH 0564/1851] v4l: xilinx-gamma: Make max-width and max-height
 mandatory dt properties

commit ad7ebc1f4952a8b8aa93cbb6b9ae2eea74d1724b from
https://github.com/Xilinx/linux-xlnx.git

This patch makes the xlnx,max-width and xlnx,max-height mandatory dt
properties. As this breaks backward compatibility, a warning message is
also added.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-gamma.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-gamma.c b/drivers/media/platform/xilinx/xilinx-gamma.c
index d113fca9b624..d7996d0f34fa 100644
--- a/drivers/media/platform/xilinx/xilinx-gamma.c
+++ b/drivers/media/platform/xilinx/xilinx-gamma.c
@@ -28,8 +28,6 @@
 #define XGAMMA_MIN_WIDTH	(64)
 #define XGAMMA_MAX_WIDTH	(8192)
 #define XGAMMA_DEF_WIDTH	(1280)
-#define XGAMMA_DEF_MAX_HEIGHT	(2160)
-#define XGAMMA_DEF_MAX_WIDTH	(3840)
 
 #define XGAMMA_AP_CTRL			(0x0000)
 #define XGAMMA_GIE			(0x0004)
@@ -360,7 +358,8 @@ static int xg_parse_of(struct xgamma_dev *xg)
 
 	rval = of_property_read_u32(node, "xlnx,max-height", &xg->max_height);
 	if (rval < 0) {
-		xg->max_height = XGAMMA_DEF_MAX_HEIGHT;
+		dev_err(dev, "xlnx,max-height is missing!");
+		return -EINVAL;
 	} else if (xg->max_height > XGAMMA_MAX_HEIGHT ||
 		   xg->max_height < XGAMMA_MIN_HEIGHT) {
 		dev_err(dev, "Invalid height in dt");
@@ -369,7 +368,8 @@ static int xg_parse_of(struct xgamma_dev *xg)
 
 	rval = of_property_read_u32(node, "xlnx,max-width", &xg->max_width);
 	if (rval < 0) {
-		xg->max_width = XGAMMA_DEF_MAX_WIDTH;
+		dev_err(dev, "xlnx,max-width is missing!");
+		return -EINVAL;
 	} else if (xg->max_width > XGAMMA_MAX_WIDTH ||
 		   xg->max_width < XGAMMA_MIN_WIDTH) {
 		dev_err(dev, "Invalid width in dt");
-- 
2.31.1

